website/docs/common-scripts/xudt.mdx

```diff
@@ -0,0 +1,346 @@
+---
+id: xudt
+title: xUDT Script
+---
+
+# xUDT Script
+
+[Extensible UDT(xUDT)](https://github.com/nervosnetwork/ckb-production-scripts) is the User-Defined-Token(fungible token) Script implementation on CKB. When issuing tokens, most developers use xUDT as the smart contract. You can think of it as the `ERC-20` smart contract on Ethereum. There is also a [dApp tutorial](/docs/dapp/create-token) on creating fungible token using xUDT script.
+
+## How xUDT works
+
+### Data Structure for xUDT Cell
+
+An xUDT cell is backward compatible with [Simple UDT](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0025-simple-udt/0025-simple-udt.md), all the existing rules defined in the Simple UDT spec must still hold true for xUDT cells. On top of sUDT, xUDT extends a cell like the following:
+
+```yaml
+data:
+  <amount: uint128> <xUDT data>
+type:
+  code_hash: xUDT type script
+  args: <owner lock script hash> <xUDT args>
+lock: <user_defined>
+```
+
+The `amount` is a 128-bit unsigned integer in little endian format. The added `xUDT args` and `xUDT data` parts provide all the new functions needed by xUDT, the
+detailed structure is explained below.
+
+### xUDT Args
+
+xUDT args has the following structure:
+
+```
+<4-byte xUDT flags> <Variable length bytes, extension data>
+```
+
+Depending on the content of `flags`, which is represented as a 32-bit unsigned
+integer in little-endian format, different extension data might be attached:
+
+• If `flags & 0x1FFFFFFF` is 0, no extension data is required. Note a
+backward-compatible way of viewing things, which is that a plain sUDT cell also
+has a hidden `flags` field with all zeros.
+
+• If `flags & 0x1FFFFFFF` is 0x1, extension data will contain a
+[molecule](https://github.com/nervosnetwork/molecule) serialized `ScriptVec`
+structure:
+
+```
+table Script {
+    code_hash:      Byte32,
+    hash_type:      byte,
+    args:           Bytes,
+}
+
+vector ScriptVec <Script>
+```
+
+Each entry included in `ScriptVec` structure is interpreted as
+an extension script with additional behaviors. When an xUDT script is
+executed, it will run through each included extension script. Only when all
+extension scripts pass validation, will xUDT also consider the validation to be
+successful.
+
+An extension script can be loaded in any of the following ways:
```

Comment:
```suggestion
Each entry included in `ScriptVec` structure is interpreted as
an extension Script with additional behaviors. When an xUDT Script is
executed, it will run through each included extension Script. Only when all
extension Scripts pass validation will xUDT consider the validation to be
successful.

An extension Script can be loaded in any of the following ways:
```